server/text_generation_server/models/causal_lm.py

```diff
@@ -233,7 +233,7 @@ def filter(self, request_ids: List[int]) -> Optional["CausalLMBatch"]:
         ]
 
         # Ensure that past_key_values tensors can be updated in-place
-        if type(self.past_key_values[0]) == tuple:
+        if isinstance(self.past_key_values[0], tuple):
```

Comment:
No. 
type(xxX) == tuple  was definitely intended here, ti's not a rookie mistake.
It's because xxx can be an instance of tuple, yet not a tuple.

I don't remember specificially the issue at hand, but I'd rather trust our past selves than the linter here.

Reply:
ahhh great catch!!

It seems that a variable can be an instance of a `tuple` via inheritance which would return True, even when its not a base `tuple`

example repro below
```python
class CustomTuple(tuple):
    pass

ct = CustomTuple([1, 2, 3])

print(type(ct) == tuple)  # False
print(isinstance(ct, tuple))  # True
``` 

reverted `isinstance` changes and opted to replace the `==` with `is` to make the linter happy. 

**This should be equivalent, `is` checks for object identity where `==` checks for value, and in the case of built-in types this will be the same.